src/app/shared/products/products-filter/products-filter.pipe.ts

```diff
@@ -0,0 +1,11 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Product} from 'src/app/shared/products/product.interface';
+
+@Pipe({
+    name: 'productsFilter',
+})
+export class ProductsFilterPipe implements PipeTransform {
+    transform(products: Product[] | null, searchName: string): Product[] | null {
+        return products?.filter((product: Product) => product.name.includes(searchName)) || null;
```

Comment:
Просто рекомендация: обычно в подобных ситуациях используют деструктуризацию, для получения целевого значения
```suggestion
        return products?.filter(({name}) => name.includes(searchName)) || null;
```